pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -31,34 +31,23 @@ func NewDualWriterMode2(legacy LegacyStorage, storage Storage) *DualWriterMode2
 }
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode2) Create(ctx context.Context, original runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	ctx = klog.NewContext(ctx, d.Log)
 
-	created, err := d.Legacy.Create(ctx, obj, createValidation, options)
+	created, err := d.Legacy.Create(ctx, original, createValidation, options)
 	if err != nil {
 		d.Log.Error(err, "unable to create object in legacy storage")
 		return created, err
 	}
 
-	accessorCreated, err := meta.Accessor(created)
+	created, err = enrichReturnedObject(original, created, true)
 	if err != nil {
 		return created, err
 	}
 
-	accessorOld, err := meta.Accessor(obj)
-	if err != nil {
-		return created, err
-	}
-
-	enrichObject(accessorOld, accessorCreated)
-
-	// create method expects an empty resource version
-	accessorCreated.SetResourceVersion("")
-	accessorCreated.SetUID("")
-
 	rsp, err := d.Storage.Create(ctx, created, createValidation, options)
 	if err != nil {
-		d.Log.WithValues("name", accessorCreated.GetName(), "resourceVersion", accessorCreated.GetResourceVersion()).Error(err, "unable to create object in duplicate storage")
+		d.Log.WithValues("kind", options.Kind).Error(err, "unable to create object in storage")
```

Comment:
hm, so the short answer is, I don't know if we need this or not. I removed it for now since there are less errors to be checked just for the sake of a log and also because I wanted to normalise all the logs to sort all logging more or less the same thing. Also, since we will have a syncer which will have to go through rvs and names anyways to do the comparison, for now, I think it should be ok to just have an idea that, for example `35 playlists are out of sync in the last week`, which is an info we can easily get with this log.
We might decide that having this info will be critical for the log, but if that's the case I'd rather do this in a different PR where I wrap the log line around an error check for the accessor.